src/lseq/lseq.rs

```diff
@@ -348,10 +364,13 @@ impl<V: Ord + Clone + Display, A: Actor + Display> LSeq<V, A> {
             } else {
                 // there is no number for q at this depth, thus we use the maximum
                 // possible for this depth (it should be the same as arity of this depth - 1)
+
+                // (drusu) I feel you want to keep the q_position "exclusive" so that taking difference gives you
+                //         a length, ie. don't subtract 1
```

Comment:
yes, q is exclusive, but the shifting is giving as the arity which is the last position + 1, e.g. arity 4 then max pos is 3, and we want 3 to be exclusive